dockers/docker-lldp-sv2/lldpmgrd

```diff
@@ -0,0 +1,188 @@
+                                                self.REDIS_HOSTNAME,
+                                                self.REDIS_PORT,
+                                                self.REDIS_TIMEOUT_USECS)
+
+    def update_lldp_config_for_port(self, port_name):
+        """
+        For port `port_name`, look up the neighboring device's hostname and
+        corresponding port alias in the Config database, then form the
+        appropriate lldpcli configuration command and run it.
+        """
+        CMD_FMT = "lldpcli configure ports {0} lldp portidsubtype local {1} description {2}:{3}"
+        TABLE_SEPARATOR = "|"
+
+        # Retrieve all entires from the Device Neighbor table
+        device_neighbor_table = swsscommon.Table(self.config_db, swsscommon.CFG_DEVICE_NEIGHBOR_TABLE_NAME, TABLE_SEPARATOR)
+        (status, fvp) = device_neighbor_table.get(port_name)
+        if status is False:
+            log_info("Port '{}' not found in {} table in Config DB. Not updating LLDP config for this port.".format(port_name, swsscommon.CFG_DEVICE_NEIGHBOR_TABLE_NAME))
+            return
+
+        # Convert list of tuples to a dictionary
+        device_neighbor_table_dict = dict(fvp)
+
+        neighbor_hostname = device_neighbor_table_dict["name"]
+        if not neighbor_hostname:
+            log_error("Failed to obtain neighbor hostname for port '{}'".format(port_name))
+            return
+
+        neighbor_portname = device_neighbor_table_dict["port"]
+        if not neighbor_hostname:
+            log_error("Failed to obtain neighbor port name for port '{}'".format(port_name))
+            return
+
+        # Retrieve all entires from the Port table
+        port_table = swsscommon.Table(self.config_db, swsscommon.CFG_PORT_TABLE_NAME, TABLE_SEPARATOR)
+        (status, fvp) = port_table.get(port_name)
+        if status is False:
+            log_error("Port '{}' not found in {} table in Config DB.".format(port_name, swsscommon.CFG_PORT_TABLE_NAME))
+            return
+
+        # Convert list of tuples to a dictionary
+        port_table_dict = dict(fvp)
+
+        port_alias = port_table_dict["alias"]
+        if not port_alias:
+            log_error("Failed to obtain port alias for port '{}'".format(port_name))
+            return
+
+        cmd = CMD_FMT.format(port_name, port_alias, neighbor_hostname, neighbor_portname)
+
+        log_info("Running command: '{}'".format(cmd))
+
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        (stdout, stderr) = proc.communicate()
+
+        if proc.returncode != 0:
+            log_error("Error running command '{}': {}".format(cmd, stderr))
+
+    def run(self):
+        """
+        Infinite loop. Subscribes to notifications of changes in the PORT table
+        of the Redis State database. When we are notified of the creation of an
+        interface, update LLDP configuration accordingly.
```

Comment:
What about removing of an interface. Do you catch such case?

Reply:
No. We decided this was unnecessary.